pkg/ingester/errors.go

```diff
@@ -28,37 +30,52 @@ var (
 
 type validationError struct {
 	err    error // underlying error
-	code   int
+	status *status.Status
 	labels labels.Labels
 }
 
-func makeLimitError(err error) error {
-	return &validationError{
-		err:  err,
-		code: http.StatusBadRequest,
+func newValidationError(err error, code int) validationError {
+	return validationError{
```

Comment:
why removing the address operator and switching to non-pointers?

Reply:
Because I have also replaced 
```
func (e *validationError) Error() string
```
with
```
func (e validationError) Error() string
```

But I will revert this for the `validationError` type

Reply:
Yes, I've seen that, I was just wondering why. I don't mind the change, if there was a good reason for it.